source/type/message.h

```diff
@@ -411,9 +411,9 @@ class DisruptionHolder {
 struct ImpactedVJ {
     const VehicleJourney* vj;  // vj before impact
     ValidityPattern new_vp;
-    std::set<StopPoint*> impacted_stops;
-    ImpactedVJ(const VehicleJourney* vj, ValidityPattern vp, std::set<StopPoint*> r)
-        : vj(vj), new_vp(vp), impacted_stops(std::move(r)) {}
+    std::set<uint16_t> impacted_ranks;
```

Comment:
For compile check, it would be nice to type it more (everywhere used), like what's done on `Idx` in https://github.com/CanalTP/utils/blob/f86e6eb15c0d0e84cae8cd6541239592fb5daae4/idx_map.h#L39-L56
```suggestion
    std::set<Rank<StopTime>> impacted_ranks;
```
Or maybe just typedef for clarity (especially in get_sections_ranks where different kind of indexes are mixed) ?
```suggestion
    std::set<stop_time_rank_t> impacted_ranks;
```

----

Still, we could even end-up mixing rank from base-VJ and the ones from rt-VJ... But I don't see what we can do about that :man_shrugging:

Reply:
Like we said on Riot didn't have time to do it sorry :(. But feel free to do it.

Reply:
Strong-typing it is an important job, worth separating PR.
I will try to open a separate one after merge of yours ;)